askama_parser/src/error.rs

```diff
@@ -0,0 +1,46 @@
+use nom::error::{Error, ErrorKind, FromExternalError, ParseError};
+use std::borrow::Cow;
+
+#[derive(Debug)]
+pub(crate) struct ParsingError<I> {
+    pub(crate) input: I,
+    pub(crate) message: Option<Cow<'static, str>>,
+}
+
+impl<I> ParseError<I> for ParsingError<I> {
+    fn from_error_kind(input: I, _code: ErrorKind) -> Self {
+        Self {
+            input,
+            message: None,
+        }
+    }
+
+    fn append(_: I, _: ErrorKind, other: Self) -> Self {
+        other
+    }
+}
+
+impl<I, E: std::fmt::Display> FromExternalError<I, E> for ParsingError<I> {
+    fn from_external_error(input: I, _kind: ErrorKind, e: E) -> Self {
+        Self {
+            input,
+            message: Some(Cow::Owned(e.to_string())),
+        }
+    }
+}
+
+impl<I> ParsingError<I> {
+    pub(crate) fn from_err(error: nom::Err<Error<I>>) -> nom::Err<Self> {
```

Comment:
Can this be a `From` or `Into` impl instead, or does that run into coherence issues?

Reply:
`From` forces to return `Self`, which is not the case here, hence why I didn't implement it.

Reply:
Ah, that makes sense.